controllers/release/release_controller.go

```diff
@@ -69,7 +70,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	adapter := NewAdapter(release, logger, r.Client, ctx)
+	adapter := NewAdapter(release, logger, r.Client, ctx, req.ClusterName)
```

Comment:
Ok, not sure this is quite right. I think we can improve the code by not passing the clusterName here. We can replace this line with the following:
```
if req.ClusterName != "" {
  ctx = logicalcluster.WithCluster(context, logicalcluster.New(cluster))
}
adapter := NewAdapter(release, logger, r.Client, ctx)
```

Then, all the instances that were using the context beforehand, can keep the same code.

For those places in which you actually need to get the name of the cluster, you can use the following:
```
clusterName, ok := logicalcluster.ClusterFromContext(a.context)
if ok { // We are running on KCP
  // whatever
}
```

I haven't tested any of this though, so give it a go yourself.

Reply:
As discussed only, the context passed here would target the `dev` workspace always as we are reconciling a Release that is always created in the `dev` workspace. For the `managed` one, we should add another context to the adapter that is set once the ReleaseLink is loaded and we know the name of the workspace. If the Workspace field is not set, we should reference the dev context.

Reply:
Done